psdk_wrapper/src/psdk_wrapper.cpp

```diff
@@ -87,30 +88,57 @@ PSDKWrapper::PSDKWrapper(const std::string &node_name)
-  psdk_ros2::global_telemetry_ptr_ = telemetry_module_;
-  camera_module_ = std::make_shared<CameraModule>("camera_node");
-  psdk_ros2::global_camera_ptr_ = camera_module_;
-  liveview_module_ = std::make_shared<LiveviewModule>("liveview_node");
-  psdk_ros2::global_liveview_ptr_ = liveview_module_;
-  gimbal_module_ = std::make_shared<GimbalModule>("gimbal_node");
-  hms_module_ = std::make_shared<HmsModule>("hms_node");
-  psdk_ros2::global_hms_ptr_ = hms_module_;
-  perception_module_ = std::make_shared<PerceptionModule>("perception_node");
-  psdk_ros2::global_perception_ptr_ = perception_module_;
-
-  // Start the threads
-  telemetry_thread_ = std::make_unique<utils::NodeThread>(telemetry_module_);
-  flight_control_thread_ =
-      std::make_unique<utils::NodeThread>(flight_control_module_);
-  camera_thread_ = std::make_unique<utils::NodeThread>(camera_module_);
-  liveview_thread_ = std::make_unique<utils::NodeThread>(liveview_module_);
-  gimbal_thread_ = std::make_unique<utils::NodeThread>(gimbal_module_);
-  hms_thread_ = std::make_unique<utils::NodeThread>(hms_module_);
-  perception_thread_ = std::make_unique<utils::NodeThread>(perception_module_);
+  load_parameters();
+
+  if (is_telemetry_module_mandatory_)
+  {
+    telemetry_module_ = std::make_shared<TelemetryModule>("telemetry_node");
+    psdk_ros2::global_telemetry_ptr_ = telemetry_module_;
+    telemetry_thread_ = std::make_unique<utils::NodeThread>(telemetry_module_);
+  }
+
+  if (is_flight_control_module_mandatory_)
+  {
+    flight_control_module_ =
+        std::make_shared<FlightControlModule>("flight_control_node");
+    flight_control_thread_ =
+        std::make_unique<utils::NodeThread>(flight_control_module_);
+  }
+
+  if (is_camera_module_mandatory_)
+  {
+    camera_module_ = std::make_shared<CameraModule>("camera_node");
+    psdk_ros2::global_camera_ptr_ = camera_module_;
+    camera_thread_ = std::make_unique<utils::NodeThread>(camera_module_);
+  }
+
+  if (is_gimbal_module_mandatory_)
+  {
+    gimbal_module_ = std::make_shared<GimbalModule>("gimbal_node");
+    gimbal_thread_ = std::make_unique<utils::NodeThread>(gimbal_module_);
+  }
+
+  if (is_liveview_module_mandatory_)
+  {
+    liveview_module_ = std::make_shared<LiveviewModule>("liveview_node");
+    psdk_ros2::global_liveview_ptr_ = liveview_module_;
+    liveview_thread_ = std::make_unique<utils::NodeThread>(liveview_module_);
+  }
+
+  if (is_hms_module_mandatory_)
+  {
+    hms_module_ = std::make_shared<HmsModule>("hms_node");
+    psdk_ros2::global_hms_ptr_ = hms_module_;
+    hms_thread_ = std::make_unique<utils::NodeThread>(hms_module_);
+  }
+
+  if (is_perception_module_mandatory_)
+  {
+    perception_module_ = std::make_shared<PerceptionModule>("perception_node");
+    psdk_ros2::global_perception_ptr_ = perception_module_;
+    perception_thread_ =
+        std::make_unique<utils::NodeThread>(perception_module_);
+  }
```

Comment:
I think this can be optimized using templated function. Let's discuss it offline as this is a comment I have for most of the changes you have added.